junit-platform-engine/src/main/java/org/junit/platform/engine/UniqueId.java

```diff
@@ -81,16 +83,23 @@ public static UniqueId root(String segmentType, String value) {
 	}
 
 	private final UniqueIdFormat uniqueIdFormat;
-	private final List<Segment> segments = new ArrayList<>();
+	private final List<Segment> segments;
 
 	private UniqueId(UniqueIdFormat uniqueIdFormat, Segment segment) {
 		this.uniqueIdFormat = uniqueIdFormat;
-		this.segments.add(segment);
+		this.segments = singletonList(segment);
 	}
 
+	/**
+	 * Initialize an {@code UniqueId} instance.
+	 *
+	 * @implNote A defensive copy of the segment list is <b>not</b> created by
+	 * this implementation. All callers should drop the list instance immediately
+	 * they pass into this constructor.
```

Comment:
--> All callers should immediately drop the reference to the list instance that they pass into this constructor.